Add documentation for properties in TensorIterator.

ezyang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #39800 Eliminate TensorIterator::add_output with explicit dtype.
* #39798 Delete redundant device/dtype in TensorIterator add_input/add_output
* **#39792 Add documentation for properties in TensorIterator.**
* #39789 Separate "configuration" properties in TensorIterator

I also deleted the dead TensorIterator::remove_dimension,
and reordered some properties so they were more logically
grouped.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D21981739](https://our.internmc.facebook.com/intern/diff/D21981739)

ezyang committed:
Add documentation for properties in TensorIterator.

I also deleted the dead TensorIterator::remove_dimension,
and reordered some properties so they were more logically
grouped.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

[ghstack-poisoned]

mruberry reviewed (approved):
A couple nits. Always nice to have more comments.

ezyang committed:
Update on "Add documentation for properties in TensorIterator."


I also deleted the dead TensorIterator::remove_dimension,
and reordered some properties so they were more logically
grouped.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

[ghstack-poisoned]
